addons/project_hr_expense/models/project_project.py

```diff
@@ -72,25 +72,35 @@ def _get_expenses_profitability_items(self, with_action=True):
             return {}
         can_see_expense = with_action and self.env.user.has_group('hr_expense.group_hr_expense_team_approver')
 
-        expenses_read_group = self.env['hr.expense']._read_group(
-            [
+        expenses = self.env['hr.expense'].search_fetch(
+            domain=[
                 ('sheet_id.state', 'in', ['post', 'done']),
                 ('analytic_distribution', 'in', self.account_id.ids),
             ],
-            groupby=['currency_id'],
-            aggregates=['id:array_agg', 'untaxed_amount_currency:sum'],
+            field_names=['currency_id', 'untaxed_amount_currency', 'analytic_distribution']
         )
-        if not expenses_read_group:
+        if not expenses:
             return {}
         expense_ids = []
         amount_billed = 0.0
-        for currency, ids, untaxed_amount_currency_sum in expenses_read_group:
+        for expense in expenses:
+            analytic_contribution = (
+                sum(
+                    percentage
+                    for ids, percentage in expense.analytic_distribution.items()
+                    if str(self.account_id.id) in ids.split(",")
+                )
+                / 100
+            )
             if can_see_expense:
-                expense_ids.extend(ids)
-            amount_billed += currency._convert(
-                from_amount=untaxed_amount_currency_sum,
-                to_currency=self.currency_id,
-                company=self.company_id,
+                expense_ids.extend(expense.ids)
+            amount_billed += (
+                expense.currency_id._convert(
+                    from_amount=expense.untaxed_amount_currency,
+                    to_currency=self.currency_id,
+                    company=self.company_id,
+                )
+                * analytic_contribution
```

Comment:
```suggestion
                ) * analytic_contribution
```
I don't see a reason to wrap this here